app/src/main/java/org/mozilla/fenix/tabstray/inactivetabs/InactiveTabs.kt

```diff
@@ -0,0 +1,355 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.compose.list.ExpandableListHeader
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+private val ROUNDED_CORNER_SHAPE = RoundedCornerShape(8.dp)
+
+/**
+ * Top-level Composable for displaying an expandable section of Inactive Tabs
+ *
+ * @param inactiveTabs List of [InactiveTabListItem] to display.
+ * @param expanded Boolean to determine whether to show the inactive tabs section expanded or collapsed.
+ * @param showAutoCloseDialog Boolean to determine whether to show the auto close inactive tabs dialog.
+ * @param onHeaderClick Lambda for handling clicks on the inactive tabs section header.
+ * @param onDeleteAllClick Lambda for handling clicks on the delete all inactive tabs button.
+ * @param onAutoCloseDismissClick Lambda for handling clicks on the auto close dialog's dismiss button.
+ * @param onEnableAutoCloseClick Lambda for handling clicks on the auto close dialog's enable button.
+ * @param onTabClick Lambda for handling clicks on a specific inactive tab.
+ * @param onTabCloseClick Lambda for handling clicks on a specific inactive tab's close button.
+ */
+@Composable
+@Suppress("LongParameterList")
+fun InactiveTabsList(
+    inactiveTabs: List<InactiveTabListItem>,
+    expanded: Boolean,
+    showAutoCloseDialog: Boolean,
+    onHeaderClick: () -> Unit,
+    onDeleteAllClick: () -> Unit,
```

Comment:
Another potential improvement could be: `onDeleteAllClick: (List<String>) -> Unit`. During manual testing I broke the delete button by the way I hacked in an inactive tab, which could've been avoided if the underlying deletion implementation was more written in a more functional style. Likely out of scope for now though

Reply:
Could you elaborate more on how you hacked in an inactive tab?

Reply:
Guess I should've saved the stash 😅 I think I just added a tab to `TabsTrayStore.state.inactiveTabs` in `TabsTrayFragment`. IIRC I had to manipulate `List<TabSessionState>.getInactiveTabs`. That at least got me a tab, but since it wasn't originally from `browserStore.state` the two got out of sync when I tried the delete all button.

I don't think there's any real risk of something like that happening in prod though, I just generally think relying on global state like in `TabsTrayController::handleDeleteAllInactiveTabs` can be a bit of an anti-pattern. Super out-of-scope for this ticket though